src/pmp/visualization/PhongShader.h

```diff
@@ -9,121 +9,130 @@
 #ifndef __EMSCRIPTEN__
-    "#version 330\n"
+    "#version 330"
 #else
-    "#version 300 es\n"
+    "#version 300 es"
 #endif
-    "\n"
-    "layout (location=0) in vec4 v_position;\n"
-    "layout (location=1) in vec3 v_normal;\n"
-    "layout (location=2) in vec2 v_tex;\n"
-    "out vec3 v2f_normal;\n"
-    "out vec2 v2f_tex;\n"
-    "out vec3 v2f_view;\n"
-    "uniform mat4 modelview_projection_matrix;\n"
-    "uniform mat4 modelview_matrix;\n"
-    "uniform mat3 normal_matrix;\n"
-    "uniform float point_size;\n"
-    "uniform bool show_texture_layout;\n"
-    "\n"
-    "void main()\n"
-    "{\n"
-    "   v2f_normal  = normal_matrix * v_normal;\n"
-    "   v2f_tex     = v_tex;\n"
-    "   vec4 pos    = show_texture_layout ? vec4(v_tex, 0.0, 1.0) : v_position;\n"
-    "   v2f_view    = -(modelview_matrix * pos).xyz;\n"
-    "   gl_PointSize = point_size;\n"
-    "   gl_Position = modelview_projection_matrix * pos;\n"
-    "} \n";
-
-
-static const char* phong_fshader =
+R"glsl(
+layout (location=0) in vec4 v_position;
+layout (location=1) in vec3 v_normal;
+layout (location=2) in vec2 v_tex;
+
+out vec3 v2f_normal;
+out vec2 v2f_tex;
+out vec3 v2f_view;
+
+uniform mat4 modelview_projection_matrix;
+uniform mat4 modelview_matrix;
+uniform mat3 normal_matrix;
+uniform float point_size;
+uniform bool show_texture_layout;
+
+void main()
+{
+    v2f_normal   = normal_matrix * v_normal;
+    v2f_tex      = v_tex;
+    vec4 pos     = show_texture_layout ? vec4(v_tex, 0.0, 1.0) : v_position;
+    v2f_view     = -(modelview_matrix * pos).xyz;
+    gl_PointSize = point_size;
+    gl_Position  = modelview_projection_matrix * pos;
+}
+)glsl";
+
+
+static const char* phong_fshader = 
 #ifndef __EMSCRIPTEN__
-    "#version 330\n"
+    "#version 330"
 #else
-    "#version 300 es\n"
-    "precision mediump float;\n"
+    "#version 300 es"
 #endif
-    "\n"
-    "in vec3  v2f_normal;\n"
-    "in vec2  v2f_tex;\n"
-    "in vec3  v2f_view;\n"
-    "\n"
-    "uniform bool   use_lighting;\n"
-    "uniform bool   use_texture;\n"
-    "uniform bool   use_srgb;\n"
-    "uniform vec3   front_color;\n"
-    "uniform vec3   back_color;\n"
-    "uniform float  ambient;\n"
-    "uniform float  diffuse;\n"
-    "uniform float  specular;\n"
-    "uniform float  shininess;\n"
-    "uniform float  alpha;\n"
-    "uniform vec3   light1;\n"
-    "uniform vec3   light2;\n"
-    "\n"
-    "uniform sampler2D mytexture;\n"
-    "\n"
-    "out vec4 f_color;\n"
-    "\n"
-    "void main()\n"
-    "{\n"
-    "    vec3 color = gl_FrontFacing ? front_color : back_color;\n"
-    "    vec3 rgb;\n"
-    "\n"
-    "    if (use_lighting)\n"
-    "    {\n"
-    "       vec3 L1 = normalize(light1);\n"
-    "       vec3 L2 = normalize(light2);\n"
-    "       vec3 N  = normalize(v2f_normal);\n"
-    "       vec3 V  = normalize(v2f_view);\n"
-    "       \n"
-    //"       if (!gl_FrontFacing) N = -N;\n" // gl_FrontFacing does not work on Mario's new MacBook
```

Comment:
Dead code should be removed

Reply:
this code will be removed anyway. the new one does not have dead lines.

Reply:
Of course, my bad.